### PR TITLE
[MKT-458]: feat/during blackFriday page

### PR DESCRIPTION
**Pull request by jaaaaavier**
- JSONs Modified, for new content
- New Background gradient
- New CTA design
- New HeroSection Desingn for mobile
- Redirection of the ButtonDeal
- Mobile gap for countdown
- Remove TopBanner
- New Nabvar behavior for when scroll and is black friday
- PriceCard desingn for darkmode
- PricingSection desingn for darkmode
- PricingSectionWrapper desingn for darkmode
- PlanSelector desingn for darkmode
- Switch desingn for darkmode
- BlackFriday Page new Design
- Addign coupons to business, family and lifetime page

### CODE REVIEW

File: src/assets/lang/en/priceCard.json [hunk 1]
```diff
@@ -122,6 +122,7 @@
     },
     "individuals": {
       "200GB": [
+        "200GB encrypted storage",
```
Comment (xabg2) [inline src/assets/lang/en/priceCard.json:125]: Be careful, you added the storage only in English. Is it correct?

File: src/components/black-friday/CtaSection.tsx [hunk 2]
```diff
@@ -7,36 +7,52 @@ import Image from 'next/image';
...
-      <div className={`relative flex flex-col lg:flex-row items-center justify-center space-y-10 lg:space-y-0 lg:space-x-10 p-5 sm:p-10 rounded-2xl shadow-lg ${styles.radialGradient}`}>
-        <div className="flex justify-center lg:flex">
-          <Image
-            src={getImage('/images/black-friday/internxt_BF_1.webp')}
-            height={219}
-            width={277}
-            alt="Hourglass"
-          />
+    <section className="overflow-hidden bg-highlight p-5 sm:p-10">
+      {/* Versión Móvil */}
```
Comment (xabg2) [inline src/components/black-friday/CtaSection.tsx]: Remove this comment

File: src/components/black-friday/CtaSection.tsx [hunk 3]
```diff
@@ -7,36 +7,52 @@ import Image from 'next/image';
...
+        </div>
+        <Image
+          src={getImage('/images/black-friday/internxt_black_friday_2024.webp')}
+          height={219}
+          width={277}
+          alt="Black Friday Discounts"
+        />
+      </div>
+
+      {/* Versión de Escritorio */}
```
Comment (xabg2) [inline src/components/black-friday/CtaSection.tsx]: Same

File: src/components/black-friday/HeroSection.tsx [hunk 4]
```diff
@@ -1,35 +1,43 @@
...
+              </div>
             </div>
           </div>
-          <div className="flex w-full  flex-col md:mt-0 md:max-w-none md:flex-row">
+          <div className="flex w-full flex-col md:mt-0 md:max-w-none md:flex-row">
             <div className="relative flex items-center justify-center md:h-[600px] lg:left-16 lg:mt-10">
-              <Image src={getImage('/images/black-friday/internxt_black_friday_2024.webp')} alt={'BFCampaign'} width={520} height={520} />
+              <Image
+                src={getImage('/images/black-friday/internxt_black_friday_2024.webp')}
+                alt={'BFCampaign'}
```
Comment (xabg2) [inline src/components/black-friday/HeroSection.tsx]: Use better alt name

File: src/components/layout/Layout.tsx [hunk 5]
```diff
@@ -59,7 +59,7 @@ LayoutProps) {
   const { dialogIsOpen } = useGlobalDialog();
   const pathname = pathnameForSEO ? pathnameForSEO : router.pathname === '/' ? '' : router.pathname;
   const lang = router.locale;
- const shouldShowBanner = pathname === '/black-friday' && dialogIsOpen(GlobalDialog.TopBanner);
+  const shouldShowBanner = pathname === '/' && dialogIsOpen(GlobalDialog.TopBanner);
```
Comment (xabg2) [inline src/components/layout/Layout.tsx]: Only for the home page?

File: src/components/shared/pricing/components/PlanSelector.tsx [hunk 6]
```diff
@@ -9,43 +9,45 @@ interface PlanSwitchProps {
   hidePlanSelectorComponent?: boolean;
   hideBusinessSelector?: boolean;
   isMonthly?: boolean;
-  isHalloween?: boolean;
+  darkMode?: boolean;
   onPlanTypeChange: (activeSwitchPlan: string, billedFrequency?: Interval) => void;
 }
-
+const getBackgroundColorClass = (darkMode: boolean) => (darkMode ? 'bg-gray-90' : 'bg-cool-gray-10');
```
Comment (xabg2) [inline src/components/shared/pricing/components/PlanSelector.tsx]: Is this used?

File: src/pages/black-friday.tsx [hunk 7]
```diff
@@ -38,72 +55,114 @@ const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang, footerL
       description: langJson.blackFriday.FeatureSection.feature4.subtitle1,
     },
   ];
+
+  const onBusinessPlansSelected = (isBusiness: boolean) => {
+    setIsBusiness(isBusiness);
+  };
+
+  const onCheckoutButtonClicked = (priceId: string, isCheckoutForLifetime: boolean) => {
+    const lifetimeSpacePlan = products?.individuals[Interval.Lifetime].find((product) => product.priceId === priceId);
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Different coupon codes are being used here for lifetime plans - is that the intention?

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/components/layout/navbars/Navbar.tsx [hunk 8]
```diff
@@ -391,7 +403,6 @@ export default function Navbar(props: Readonly<NavbarProps>) {
                       >
                         {props.textContent.links.login}
                       </a>
-                      <LanguageMobileBox darkMode={props.darkMode} singlesDay={props.singlesDay} />
```
Comment (xabg2) [inline src/components/layout/navbars/Navbar.tsx:394]: You removed the language selector for mobile view. Is it ok?

File: src/components/shared/pricing/PricingSection.tsx [hunk 9]
```diff
@@ -72,17 +73,17 @@ export const PricingSection = ({
...
   const isBusiness = activeSwitchPlan === 'Business';
-  const labelDiscount = isBusiness ? '10' : '23';
+  const labelDiscount = '15';
   const showLoadingCards = loadingCards;
   const showBusinessCards = isBusiness && !loadingCards && !!businessBillingFrequency;
   const isIndividual = activeSwitchPlan === 'Individuals' || activeSwitchPlan === 'Lifetime';
   const showIndividualCards = isIndividual && !loadingCards;
-
-  const showSwitchComponent = activeSwitchPlan === 'Business' && !hideBusinessCards;
+  const showSwitchComponent = !loadingCards && !hideSwitchSelector && activeSwitchPlan !== 'Lifetime';
```
Comment (xabg2) [inline src/components/shared/pricing/PricingSection.tsx]: Same here.

File: src/pages/black-friday.tsx [hunk 10]
```diff
@@ -1,21 +1,37 @@
...
+import { PromoCodeName } from '@/lib/types';
+import { Interval, stripeService } from '@/components/services/stripe.service';
+import usePricing from '@/hooks/usePricing';
 import { Eye, Fingerprint, LockKey, ShieldCheck } from '@phosphor-icons/react';
+import { TextAndCardsGroupColumnSection } from '@/components/shared/components/TextAndCardsGroupColumnSection';
 
 const BLACK_FRIDAY_METATAG_ID = 'black-friday';
 
-const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang, footerLang}) => {
+const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang, footerLang, pricingLang }) => {
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Type the props using an interface

File: src/pages/black-friday.tsx [hunk 11]
```diff
@@ -1,21 +1,37 @@
...
   const metatags = metatagsDescriptions.filter((desc) => desc.id === BLACK_FRIDAY_METATAG_ID);
+  const [isBusiness, setIsBusiness] = useState(false);
+
+  const {
+    products,
+    loadingCards,
+    currencyValue,
+    coupon: individualCoupon,
+    businessCoupon,
+    lifetimeCoupons,
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Do you need the lifetime coupons?

File: src/pages/black-friday.tsx [hunk 12]
```diff
@@ -38,72 +54,109 @@ const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang, footerL
       description: langJson.blackFriday.FeatureSection.feature4.subtitle1,
     },
   ];
+
+  const onBusinessPlansSelected = (isBusiness) => {
+    setIsBusiness(isBusiness);
+  };
+
+  const onCheckoutButtonClicked = (priceId, isCheckoutForLifetime) => {
+    const lifetimeSpacePlan = products?.individuals[Interval.Lifetime].find((product) => product.priceId === priceId);
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Do you need that? Or you will use the same coupon code for all plans (Individuals | Lifetime | Business)

File: src/pages/black-friday.tsx [hunk 13]
```diff
@@ -38,72 +54,109 @@ const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang, footerL
...
+        loadingCards={loadingCards}
+        onBusinessPlansSelected={onBusinessPlansSelected}
+        onCheckoutButtonClicked={onCheckoutButtonClicked}
+        lifetimeCoupons={lifetimeCoupons}
+        hideFreeCard
+        darkMode
+        backgroundColorComponent="bg-highlight"
+        hideTitle
+        decimalDiscount={{
+          individuals: individualCoupon?.percentOff && 100 - individualCoupon?.percentOff,
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Extract the operation to a constant and use it for individuals, lifetime and business. Something like: `const decimalDiscount = individualCoupon?.percentOff && 100 - individualCoupon?.percentOff,`

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/components/layout/navbars/Navbar.tsx (hunk 8 above)
Comment (jaaaaavier) [reply src/components/layout/navbars/Navbar.tsx:394]: No, fixed

File: src/pages/black-friday.tsx (hunk 10 above)
Comment (jaaaaavier) [reply src/pages/black-friday.tsx]: Okay

File: src/pages/black-friday.tsx (hunk 13 above)
Comment (jaaaaavier) [reply src/pages/black-friday.tsx]: Okay

File: src/pages/black-friday.tsx (hunk 12 above)
Comment (jaaaaavier) [reply src/pages/black-friday.tsx]: No, i will use the same

File: src/pages/black-friday.tsx (hunk 11 above)
Comment (jaaaaavier) [reply src/pages/black-friday.tsx]: No

File: src/components/shared/pricing/PricingSection.tsx (hunk 9 above)
Comment (jaaaaavier) [reply src/components/shared/pricing/PricingSection.tsx]: I understood that the monthly plans also had to be shown in individuals. But not

File: src/pages/index.tsx [hunk 14]
```diff
@@ -83,24 +83,31 @@ const HomePage = ({ metatagsDescriptions, textContent, lang, navbarLang, footerL
...
     const planType = isBusiness ? 'business' : 'individual';
 
     stripeService.redirectToCheckout(priceId, currencyValue, planType, isCheckoutForLifetime, couponCodeForCheckout);
   };
 
+  function useIsMobile() {
+    const [isMobile, setIsMobile] = useState(false);
+
+    useEffect(() => {
+      const checkIsMobile = () => setIsMobile(window.innerWidth < 768); // Ajusta el valor según el tamaño móvil
```
Comment (xabg2) [inline src/pages/index.tsx]: Remove the comment

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx [hunk 15]
```diff
@@ -83,24 +83,31 @@ const HomePage = ({ metatagsDescriptions, textContent, lang, navbarLang, footerL
...
-        : individualCoupon?.name;
-
-    const couponCodeForCheckout = isBusiness ? businessCoupon?.name : couponCodeForB2CPlans;
+    const couponCodeForCheckout = individualCoupon?.name;
     const planType = isBusiness ? 'business' : 'individual';
 
     stripeService.redirectToCheckout(priceId, currencyValue, planType, isCheckoutForLifetime, couponCodeForCheckout);
   };
 
+  function useIsMobile() {
```
Comment (xabg2) [inline src/pages/index.tsx]: Create a custom hook with this. Just extract this to a useIsMobile and its useEffect to custom hook and call it here.

**Review by xabg2 — CHANGES_REQUESTED** [review]

**Review by xabg2 — APPROVED** [review]

**Pull request merged**